internal/alert/telegram.go

```diff
@@ -0,0 +1,49 @@
+package alert
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/ianmarmour/nvidia-clerk/internal/config"
+)
+
+type TelegramPayload struct {
+	ChatID string `json:"chat_id"`
+	Text   string `json:"text"`
+}
+
+//SendTelegramMessage Sends a notification message to a Telegram Webhook.
+func SendTelegramMessage(item string, config config.TelegramConfig, client *http.Client) error {
+	content := fmt.Sprintf("%s Ready for Purchase", item)
+
+	payload, err := json.Marshal(TelegramPayload{fmt.Sprintf("@%s", config.ChatID), content})
+
+	if err != nil {
+		log.Fatalln("Unable to marshal telegram payload.")
+	}
+
+	req, _ := http.NewRequest("POST", fmt.Sprintf("https://api.telegram.org/bot%s/sendMessage", config.APIKey), bytes.NewBuffer(payload))
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, _ := client.Do(req)
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 400 {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
Probably don't want to log.Fatal here, I would prefer to log.Println and then pass the error back up. I know it's not incredibly idiomatic but if a user has sms/discord/telegram if just one fails due to some outside error I would like the program to continue.